pgmpy/tests/test_models/test_FactorGraph.py

```diff
@@ -287,5 +287,20 @@ def test_check_model4(self):
         self.graph.add_edges_from([('a', phi3), ('c', phi3)])
         self.assertRaises(ValueError, self.graph.check_model)
 
+    def test_copy(self):
+        self.graph.add_nodes_from(['a', 'b', 'c'])
+        phi1 = Factor(['a', 'b'], [2, 2], np.random.rand(4))
+        phi2 = Factor(['b', 'c'], [2, 2], np.random.rand(4))
+        self.graph.add_nodes_from([phi1, phi2])
+        self.graph.add_edges_from([('a', phi1), ('b', phi1), 
+                                   ('b', phi2), ('c', phi2)])
+        self.graph.add_factors(phi1, phi2)
+        graph_copy = self.graph.copy()
+        self.assertIsInstance(graph_copy, FactorGraph)
+        self.assertTrue(graph_copy.check_model())
+        self.assertListEqual(graph_copy.factors[0].scope(), ['a', 'b'])
+        self.assertListEqual(graph_copy.factors[1].scope(), ['b', 'c'])
```

Comment:
@Pratik151 Add more tests. Try to change  something in the actual object and check if it also gets changed in the copy.